accepted/00052-new-package-predownloading.md

```diff
@@ -0,0 +1,114 @@
+- Feature Name: new-package-predownloading
+- Start Date: 29-04-2019
+- RFC PR: https://github.com/uyuni-project/uyuni-rfc/pull/5
+
+# Summary
+Change the current package predownloading ("staging") feature for minions. Allow users to customize the start download time on a per-Action basis.
+
+# Motivation
+
+Package pre-downloading is an essential feature to shorten the length of patch windows and is one of the features to enable bandwidth management for installations that might not have good connectivity between systems and Proxies, or Proxies and Server.
+
+Currently package pre-downloading is a global setting - either all or none of the Actions get it. Moreover the time window is unique ([documentation reference](https://github.com/SUSE/doc-susemanager/blob/a5a5c8ec/modules/reference/pages/admin/organizations.adoc#organization-details--configuration)), and a restart of Tomcat and Taskomatic is required to change it.
+
+This RFC is about adding "download only" package installation/patch application Actions that can be scheduled as needed.
+
+Motivation summary:
+ - users might want to change the download schedule depending on physical locations, often represented by System Groups
+   - this is particularly true in SUSE Manager for Retail use cases
+ - users might need to change the download schedule depending on workload, available calendar time, other factors
+ - we want users to be able to change the download schedule without restarting Tomcat and Taskomatic (currently the case as parameters are in `rhn.conf`)
+
+# Detailed design
+
+## User interface
+
+### GUI
+
+Whenever a package installation or patch application is scheduled on a minion, users will have the option to choose an advance download time:
+
+![Screenshot of the new option form](images/00052-earliest.png)
+
+*Note: the new form controls are highlighted in yellow.*
+
+Clicking on the selection element will bring up possible options:
+
+![Screenshot of the new option, selected](images/00052-selected.png)
+
+
+If an Action Chain is selected as the scheduling option, the new form elements will be disabled:
```

Comment:
This comment is more an implementation detail for later probably, but I think disabling the option is not really necessary if we design it clearly that the "Download packages" drop-down field belongs to the "Earliest" option only.
It can also be set to disabled in that case, I agree and I would keep it, but I see this as a *plus*.
Below I hacked your picture to give you a rough idea (it would needs some refinement anyway):
![pre-download](https://user-images.githubusercontent.com/7080830/57257831-e1d96880-705a-11e9-82b2-859708937947.png)

Reply:
We will surely involve you guys when the front-end writing time comes :wink: